test/test_custom_ops.py

```diff
@@ -1534,6 +1534,179 @@ def foo_impl(x):
             self.assertEqual(result.shape, ())
 
 
+class MiniOpTest(CustomOpTestCaseBase):
+    test_ns = "mini_op_test"
+
+    def _op_with_incorrect_schema(self, name):
+        lib = self.lib()
+        lib.define(f"{name}(Tensor x) -> Tensor")
+        qualname = f"{self.test_ns}::{name}"
+        lib.impl(name, lambda x: x[:], "CompositeExplicitAutograd")
+        return self.get_op(qualname)
+
+    def _op_delayed_backward_error(self, name):
+        lib = self.lib()
+        lib.define(f"{name}(Tensor x) -> Tensor")
+        qualname = f"{self.test_ns}::{name}"
+        lib.impl(name, lambda x: x.clone(), "CompositeExplicitAutograd")
+        op = self.get_op(qualname)
+
+        class Op(torch.autograd.Function):
+            @staticmethod
+            def forward(ctx, x):
+                with torch._C._AutoDispatchBelowAutograd():
+                    return op(x)
+
+            @staticmethod
+            def backward(ctx, grad):
+                raise NotImplementedError()
+
+        def autograd_impl(x):
+            return Op.apply(x)
+
+        lib.impl(name, autograd_impl, "Autograd")
+        return op
+
+    def _op_with_no_abstract_impl(self, name):
+        lib = self.lib()
+        lib.define(f"{name}(Tensor x) -> Tensor")
+        qualname = f"{self.test_ns}::{name}"
+        lib.impl(name, lambda x: x.clone(), "CPU")
+        return self.get_op(qualname)
+
+    def test_mm(self):
+        x = torch.randn(2, 3, requires_grad=True)
+        y = torch.randn(3, 5)
+        result = torch.ops.aten.mm.default(x, y)
+        self.assertEqual(result, x @ y)
+
+    def test_mm_errors(self):
+        x = torch.randn(2, 3, requires_grad=True)
+        y = torch.randn(4, 5)
+        with self.assertRaisesRegex(RuntimeError, "cannot be multiplied"):
+            result = torch.ops.aten.mm.default(x, y)
+
+    def test_nonzero(self):
+        x = torch.tensor([0, 1, 2, 0, 0])
+        y = torch.ops.aten.nonzero.default(x)
+        self.assertEqual(y, torch.tensor([[1], [2]]))
+
+    def test_inplace(self):
+        x = torch.randn(3)
+        x_clone = x.clone()
+        y = torch.ops.aten.sin_(x)
+        self.assertEqual(x, x_clone.sin())
+
+    def test_incorrect_schema(self):
+        op = self._op_with_incorrect_schema("incorrect_schema")
+        x = torch.randn(3)
+        op(x)
+
+    def test_no_abstract(self):
+        op = self._op_with_no_abstract_impl("no_abstract")
+        x = torch.randn(3)
+        op(x)
+
+    def test_delayed_error(self):
+        op = self._op_delayed_backward_error("delayed_error")
+        x = torch.randn([], requires_grad=True)
+        y = op(x)
+        with self.assertRaises(NotImplementedError):
+            y.sum().backward()
+
+    def test_delayed_error_no_requires_grad(self):
+        op = self._op_delayed_backward_error("delayed_error")
+        x = torch.randn([])
+        y = op(x)
+
+
+mini_op_test_failures_dict = {
```

Comment:
Help wanted: I don't think it is technically feasible to develop a tool that autogenerates this list. Some tests may need to be marked as "skip" if:
- a test might crash (e.g. the FakeTensor test)
- a test might be flaky (due to randomness, or bugs in PyTorch subsystems)
but it is difficult to detect the above situations.

Reply:
This would be so much easier if we were in OSS lol

Reply:
If you want to go all in on fbcode, xfail generation is automatically done by testx. So you technically don't have to do anything, except when people want to perma-suppress a test because it is definitely not a bug and just a framework limitation that can't be worked around

Reply:
Testx only does xfails per-test, while each test here may test one or more operators